packages/wallets/src/hooks/__tests__/useSyncLocalStorageClientAccounts.spec.ts

```diff
@@ -0,0 +1,195 @@
+    useLocalStorage: jest.fn((key: string) => {
+        return [
+            global.localStorage.getItem(key),
+            (data: unknown) => {
+                const stringifiedData = JSON.stringify(data);
+                global.localStorage.setItem(key, stringifiedData);
+            },
+        ];
+    }),
+    useReadLocalStorage: jest.fn((key: string) => global.localStorage.getItem(key)),
+}));
+
+jest.mock('@deriv/api', () => ({
+    ...jest.requireActual('@deriv/api'),
+    useActiveAccount: jest.fn(() => ({
+        data: {
+            account_category: 'wallet',
+            account_type: 'crypto',
+            created_at: 1701783482,
+            excluded_until: '',
+            is_disabled: false,
+            is_virtual: false,
+            landing_company_name: 'svg',
+            landing_company_shortcode: 'svg',
+            linked_to: [],
+            loginid: 'CRW1002',
+        },
+    })),
+    useSettings: jest.fn(() => ({
+        data: {
+            citizen: 'id',
+            email: 'wallet+01@deriv.com',
+            residence: 'id',
+        },
+    })),
+}));
+
+const localStorageKey = 'client.accounts';
+
+const defaultClientAccountsValue = {
+    CR2001: {
+        accepted_bch: 0,
+        account_category: 'trading',
+        account_type: 'standard',
+        balance: 250,
+        created_at: 1701783482,
+        currency: 'USD',
+        email: 'wallet+01@deriv.com',
+        excluded_until: '',
+        is_disabled: 0,
+        is_virtual: 0,
+        landing_company_name: 'svg',
+        landing_company_shortcode: 'svg',
+        linked_to: [
+            {
+                loginid: 'CRW1001',
+                platform: 'dwallet',
+            },
+        ],
+        residence: 'id',
+        token: 'a1-testR123',
+    },
+    CRW1001: {
+        accepted_bch: 0,
+        account_category: 'wallet',
+        account_type: 'doughflow',
+        balance: 9500,
+        created_at: 1701783482,
+        currency: 'USD',
+        email: 'wallet+01@deriv.com',
+        excluded_until: '',
+        is_disabled: 0,
+        is_virtual: 0,
+        landing_company_name: 'svg',
+        landing_company_shortcode: 'svg',
+        linked_to: [
+            {
+                loginid: 'CR2001',
+                platform: 'dtrade',
+            },
+        ],
+        residence: 'id',
+        token: 'a1-testF432',
+    },
+    CRW1002: {
+        accepted_bch: 0,
+        account_category: 'wallet',
+        account_type: 'crypto',
+        balance: 6,
+        created_at: 1701783482,
+        currency: 'BTC',
+        email: 'wallet+01@deriv.com',
+        excluded_until: '',
+        is_disabled: 0,
+        is_virtual: 0,
+        landing_company_name: 'svg',
+        landing_company_shortcode: 'svg',
+        linked_to: [],
+        residence: 'id',
+        token: 'a1-testQY723',
+    },
+};
+
+describe('useSyncLocalStorageClientAccounts', () => {
+    beforeEach(() => {
+        global.localStorage.clear();
```

Comment:
Typically, slightly better pattern is to completely mock local storage and completely restore original one. It makes it easier when e.g. running tests in browsers or other non-isolated environments. Though shouldn't cause an issue in our setup hopefully. 

e.g.
// before each
originalLocalStorage = global.localStorage;
global.localStorage = mockLocalStorage();
global.localStorage.setItem(...., ....)

// afterEach
global.localStorage = originalLocalStorage;

Reply:
@wojciech-deriv do you want something like this? (image attached)

<img width="853" alt="image" src="https://github.com/binary-com/deriv-app/assets/120570511/9221edae-ffbd-4189-b0a3-8e4d4723d789">

In my opinion it's too complicated and unnecessary, because I don't change anything inside of the localStorage implementation, I just add/remove items there. Not sure I can affect anything outside of the test environment. Let me know and I'll push this code to the PR if we really need this.

Reply:
yeah, something like this, just maybe separated to external file, 

its not strictly necessary, but in past I had scenarios in which cleaning local storage by running tests was annoying. E.g. I wanted to run tests in browser and those were cleaning up all my local storage keys and cookies under localhost (so all the random stuff which used to rely on localstorage / cookies, tooltips / help messages / logged in status etc was cleaned up every time I've ran tests). Its just good habit to leave global state in exactly the same state it used to be before before test.

Reply:
@wojciech-deriv OK. I've pushed the changes